datashuttle/configs.py

```diff
@@ -18,42 +18,102 @@ def __init__(self, file_path, input_dict):
+        
             utils.raise_error(
                 "remote_path must contain the full directory path with no ~ syntax"
             )
 
+        # Check SSH settings
         if self["ssh_to_remote"] is True and (
             not self["remote_host_id"] or not self["remote_host_username"]
         ):
             utils.raise_error(
                 "ssh to remote set but no remote_host_id or remote_host_username not"
-                " provided"
+                " provided."
             )
 
         if self["ssh_to_remote"] is False and (
             self["remote_host_id"] or self["remote_host_username"]
         ):
             warnings.warn(
                 "SSH to remote is false, but remote_host_id or remote_host_username"
-                " provided"
+                " provided."
             )
 
     def update_an_entry(self, option_key: str, new_info: Union[str, bool]):
         """
         Convenience function to update individual entry of configuration file.
         The config file, and currently loaded self.cfg will be updated.
 
+        Note if user updates connection type, a different remote_path
+        is used to avoid confusion. Inform the user of the new remote path.
+
+        There is always a chance that the user will make a breaking update.
+        As such set to new value, test validity and revert if breaking change.
         :param option_key: dictionary key of the option to change,
                            see make_config_file()
         :param new_info: value to update the config too
         """
-        if option_key in ["local_path", "remote_path"]:
+        original_value = copy.deepcopy(self[option_key])
+
+        if option_key in [
+            "local_path",
+            "remote_path_ssh",
+            "remote_path_local",
+        ]:
             new_info = Path(new_info)
 
         self[option_key] = new_info
-        self.dump_to_file()
+
+        change_valid = self.safe_check_current_dict_is_valid()
+
+        if change_valid:
+            self.dump_to_file()
+            utils.message_user(f"{option_key} has been updated to {new_info}")
+
+            if option_key == "ssh_to_remote":
+                if new_info:
+                    utils.message_user(
+                        f"SSH will be used to connect to project directory at: {self.get_remote_path(for_user=True)}"
+                    )
+                else:
+                    utils.message_user(
+                        f"Local filesystem will be used for project directory at: {self.get_remote_path(for_user=True)}"
+                    )
+        else:
+            self[option_key] = original_value
+            warnings.warn(f"{option_key} was not updated")
+            self[option_key] = original_value
+
+    def safe_check_current_dict_is_valid(self):
+        """ """
+        try:
+            self.check_dict_values_and_inform_user()
+            return True
+
+        except BaseException as e:  # TODO: test invalid change
```

Comment:
Tiny point, but it would be great to have TODOs as GitHub issues, so that others can see at a glance what could be worked on.

Reply:
yes agree, maybe we can have a guideline that says before making PR either move TODO to github issues or solve them. I did not make Github issue as its a very small thing, but was just been lazy!

Reply:
actually, this one was complete but I do have a bad having of leaving TODO that feel too small to be an issue but too lazy to do in the moment.

Reply:
I think anything can be an issue. If it's tiny enough so that the title is sufficient context, then I think that's fine.